src/main/scala/kamon/logback/instrumentation/AsyncAppenderInstrumentation.scala

```diff
@@ -37,6 +42,21 @@ class AsyncAppenderInstrumentation {
   @Around("execution(* ch.qos.logback.core.spi.AppenderAttachableImpl.appendLoopOnAppenders(..)) && args(event)")
   def onAppendLoopOnAppenders(pjp: ProceedingJoinPoint, event:ContextAwareLoggingEvent): Any =
     Kamon.withContext(event.getContext)(pjp.proceed())
+
+  @Before("execution(* ch.qos.logback.classic.spi.ILoggingEvent.getMDCPropertyMap()) && target(event)")
+  def beforeGetMDCPropertyMap(event: ContextAwareLoggingEvent): Unit = {
+    val context = event.getContext.get(Span.ContextKey)
+    if (context.context().traceID != IdentityProvider.NoIdentifier){
+      MDC.put(MdcTraceKey, context.context().traceID.string)
+      MDC.put(MdcSpanKey, context.context().spanID.string)
+    }
+  }
+
+  @After("execution(* ch.qos.logback.classic.spi.ILoggingEvent.prepareForDeferredProcessing()) && target(event)")
```

Comment:
I think it would be safer to turn the advice above (`beforeGetMDCPropertyMap`) into a `@Around` advice and put/remote the keys from the MDC right there.

Reply:
it may make sense, will have a look to do it that way